src/analyzer.rs

```diff
@@ -82,6 +66,41 @@ impl<'src> Analyzer<'src> {
       }
     }
 
+    let assignments = self.assignments;
+
+    AssignmentResolver::resolve_assignments(&assignments)?;
+
+    for recipe in recipes {
+      if let Some(original) = self.recipes.get(recipe.name.lexeme()) {
+        if settings.allow_duplicate_recipes {
+            self.recipes.remove(recipe.name.lexeme());
+        } else {
+          return Err(recipe.name.token().error(DuplicateRecipe {
+            recipe: original.name(),
+            first: original.line_number(),
+          }));
+        }
+      }
```

Comment:
Since insert will overwrite something already in the table, this can just be:

```suggestion
        if !settings.allow_duplicate_recipes {
          return Err(recipe.name.token().error(DuplicateRecipe {
            recipe: original.name(),
            first: original.line_number(),
          }));
        }
      }
```